From b1682941ab980031d4d9e7a276cce16f88605c67 Mon Sep 17 00:00:00 2001
From: Bartosz Golaszewski <bgolaszewski@baylibre.com>
Date: Mon, 1 Aug 2016 18:04:25 +0200
Subject: [PATCH 15/16] dt: acme: add device tree support for acme

Add the I2C multiplexer, GPIO expanders, all ina226 devices
and EEPROMs.

Signed-off-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
---
 arch/arm/boot/dts/am335x-boneblack.dts | 247 +++++++++++++++++++++++++++++++++
 1 file changed, 247 insertions(+)

diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 55c0e95..52eae9c 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -13,6 +13,217 @@
 / {
 	model = "TI AM335x BeagleBone Black";
 	compatible = "ti,am335x-bone-black", "ti,am335x-bone", "ti,am33xx";
+
+	i2cmux@1 {
+		compatible = "i2c-mux-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		mux-gpios = <&pca9534 0 GPIO_ACTIVE_LOW>;
+		i2c-parent = <&i2c1>;
+		idle-state = <1>;
+
+		i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			acme_eeprom0: acme_eeprom0@54 {
+				compatible = "at,24c32";
+				reg = <0x54>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_eeprom1: acme_eeprom1@55 {
+				compatible = "at,24c32";
+				reg = <0x55>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_eeprom2: acme_eeprom2@56 {
+				compatible = "at,24c32";
+				reg = <0x56>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_eeprom3: acme_eeprom3@57 {
+				compatible = "at,24c32";
+				reg = <0x57>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_serial0: acme_serial0@5c {
+				compatible = "at,24cs32";
+				reg = <0x5c>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_serial1: acme_serial1@5d {
+				compatible = "at,24cs32";
+				reg = <0x5d>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_serial2: acme_serial2@5e {
+				compatible = "at,24cs32";
+				reg = <0x5e>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
+			acme_serial3: acme_serial3@5f {
+				compatible = "at,24cs32";
+				reg = <0x5f>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+		};
+
+		i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			pca9535: pca9535@20 {
+				compatible = "pca9535";
+				reg = <0x20>;
+				#gpio-cells = <2>;
+			};
+
+			acme_probe0: acme_probe0@40 {
+				compatible = "ti,ina226";
+				reg = <0x40>;
+				gpio = <&pca9535 1 0>;
+			};
+
+			acme_probe_eeprom0: acme_probe_eeprom0@50 {
+				compatible = "at,24c02";
+				reg = <0x50>;
+			};
+
+			acme_probe_serial0: acme_probe_serial0@58 {
+				compatible = "at,24cs02";
+				reg = <0x58>;
+			};
+
+			acme_probe1: acme_probe1@41 {
+				compatible = "ti,ina226";
+				reg = <0x41>;
+				gpio = <&pca9535 3 0>;
+			};
+
+			acme_probe_eeprom1: acme_probe_eeprom1@51 {
+				compatible = "at,24c02";
+				reg = <0x51>;
+			};
+
+			acme_probe_serial1: acme_probe_serial1@59 {
+				compatible = "at,24cs02";
+				reg = <0x59>;
+			};
+
+			acme_probe2: acme_probe2@42 {
+				compatible = "ti,ina226";
+				reg = <0x42>;
+				gpio = <&pca9535 5 0>;
+			};
+
+			acme_probe_eeprom2: acme_probe_eeprom2@52 {
+				compatible = "at,24c02";
+				reg = <0x52>;
+			};
+
+			acme_probe_serial2: acme_probe_serial2@5a {
+				compatible = "at,24cs02";
+				reg = <0x5a>;
+			};
+
+			acme_probe3: acme_probe3@43 {
+				compatible = "ti,ina226";
+				reg = <0x43>;
+				gpio = <&pca9535 7 0>;
+			};
+
+			acme_probe_eeprom3: acme_probe_eeprom3@53 {
+				compatible = "at,24c02";
+				reg = <0x53>;
+			};
+
+			acme_probe_serial3: acme_probe_serial3@5b {
+				compatible = "at,24cs02";
+				reg = <0x5b>;
+			};
+
+			acme_probe4: acme_probe4@44 {
+				compatible = "ti,ina226";
+				reg = <0x44>;
+				gpio = <&pca9535 9 0>;
+			};
+
+			acme_probe_eeprom4: acme_probe_eeprom4@54 {
+				compatible = "at,24c02";
+				reg = <0x54>;
+			};
+
+			acme_probe_serial4: acme_probe_serial4@5c {
+				compatible = "at,24cs02";
+				reg = <0x5c>;
+			};
+
+			acme_probe5: acme_probe5@45 {
+				compatible = "ti,ina226";
+				reg = <0x45>;
+				gpio = <&pca9535 11 0>;
+			};
+
+			acme_probe_eeprom5: acme_probe_eeprom5@55 {
+				compatible = "at,24c02";
+				reg = <0x55>;
+			};
+
+			acme_probe_serial5: acme_probe_serial5@5d {
+				compatible = "at,24cs02";
+				reg = <0x5d>;
+			};
+
+			acme_probe6: acme_probe6@46 {
+				compatible = "ti,ina226";
+				reg = <0x46>;
+				gpio = <&pca9535 13 0>;
+			};
+
+			acme_probe_eeprom6: acme_probe_eeprom6@56 {
+				compatible = "at,24c02";
+				reg = <0x56>;
+			};
+
+			acme_probe_serial6: acme_probe_serial6@5e {
+				compatible = "at,24cs02";
+				reg = <0x5e>;
+			};
+
+			acme_probe7: acme_probe7@47 {
+				compatible = "ti,ina226";
+				reg = <0x47>;
+				gpio = <&pca9535 15 0>;
+			};
+
+			acme_probe_eeprom7: acme_probe_eeprom7@57 {
+				compatible = "at,24c02";
+				reg = <0x57>;
+			};
+
+			acme_probe_serial7: acme_probe_serial7@5f {
+				compatible = "at,24cs02";
+				reg = <0x5f>;
+			};
+		};
+	};
 };
 
 &ldo3_reg {
@@ -64,6 +275,20 @@
 			AM33XX_IOPAD(0x9b0, PIN_OUTPUT_PULLDOWN | MUX_MODE3)	/* xdma_event_intr0 */
 		>;
 	};
+
+	i2c1_pins: pinmux_i2c1_pins {
+		pinctrl-single,pins = <
+			0x158 (PIN_INPUT_PULLUP | MUX_MODE2)
+			0x15c (PIN_INPUT_PULLUP | MUX_MODE2)
+		>;
+	};
+
+	i2c2_pins: pinmux_i2c2_pins {
+		pinctrl-single,pins = <
+			0x178 (PIN_INPUT_PULLUP | MUX_MODE3)
+			0x17c (PIN_INPUT_PULLUP | MUX_MODE3)
+		>;
+	};
 };
 
 &lcdc {
@@ -91,6 +316,28 @@
 	};
 };
 
+&i2c1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c1_pins>;
+
+	status = "okay";
+	clock-frequency = <400000>;
+
+	pca9534: pca9534@21 {
+		compatible = "pca9534";
+		reg = <0x21>;
+		#gpio-cells = <2>;
+	};
+};
+
+&i2c2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c2_pins>;
+
+	status = "okay";
+	clock-frequency = <400000>;
+};
+
 &rtc {
 	system-power-controller;
 };
-- 
2.7.4

